apps/rush-lib/src/schemas/repo-state.schema.json

```diff
@@ -0,0 +1,18 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Rush repo-state.json file",
+  "description": "For use with the Rush tool, this file tracks the state of various features in the Rush repo. See http://rushjs.io for details.",
+
+  "type": "object",
+  "properties": {
+    "$schema": {
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+    "preferredVersionsHash": {
```

Comment:
Do we want to put the lockfile/shrinkwrap hash in this file too?

Reply:
We do! I was going to do this in a future change, but I actually encountered a bug with the shrinkwrap tamper detection when running workspaces. I've added the changes to this PR :)